File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        n = len(nums)
        count=0
        for i in range(len(nums)):
            if nums[i]==0:
                count+=1
                continue
            nums[i-count]=nums[i]
        for i in range(len(nums)-count,len(nums)):
            nums[i]=0      

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slow = 0,n=nums.size();
        for(int fast=0;fast<n;fast++){
            if(nums[fast]!=0 && nums[slow]==0)
                swap(nums[slow],nums[fast]);
            if(nums[slow]!=0)
                slow += 1;
        }
    }
};